### PR TITLE
chore: inDiet true sequence

### PR DESCRIPTION
**Pull request by JakeBalakrishna**
---if for example you have this sequence of in diet values,
true,
true,
false,
true,
---the best sequence in your diet will be 2.
---if it's,
true,
false,
true,
false,    it will be 1 etc etc etc.

### CODE REVIEW

File: src/Screens/MealOverview.tsx [hunk 1]
```diff
@@ -60,7 +62,12 @@ const styles = StyleSheet.create({
     width: "100%"
   }
 });
-//css here is kinda janky fixing it now
+//css still janky will fix in the css pr
+
+type SequenceValue = {
```
Comment (mattwilson02) [inline src/Screens/MealOverview.tsx:67]: not sure why this is typed so complexly lol, just add state as Type Meal[] and pass meals.allmeals

**Review by mattwilson02 — APPROVED** [review]

File: src/Screens/MealOverview.tsx [hunk 2]
```diff
@@ -70,6 +77,29 @@ export const MealOverview = () => {
     mealsOutDiet: state.meals.filter(item => !item.inDiet)
   }));
 
+  //chat gpt was blocked so this will be chaos lol sorry
+  const countConsecutiveTrue = (state: SequenceValue) => {
+    const allMeals = state.meals.map(
+      (item: { inDiet: boolean }) => item.inDiet
```
[the remark this answers is not in the document]
Comment (JakeBalakrishna) [reply src/Screens/MealOverview.tsx]: yea i could have removed that type declaration,  fixed

File: src/Screens/MealOverview.tsx [hunk 3]
```diff
@@ -70,6 +77,27 @@ export const MealOverview = () => {
     mealsOutDiet: state.meals.filter(item => !item.inDiet)
   }));
 
+  //chat gpt was blocked so this will be chaos lol sorry
+  const countConsecutiveTrue = (state: SequenceValue) => {
```
Comment (mattwilson02) [inline src/Screens/MealOverview.tsx:81]: ok but now you aren't using your parameter for anything in the function 🤣

Comment (mattwilson02) [reply src/Screens/MealOverview.tsx:81]: since you already have your meals in scope, do we really need a parameter

Comment (JakeBalakrishna) [reply src/Screens/MealOverview.tsx:81]: oooft

Comment (JakeBalakrishna) [reply src/Screens/MealOverview.tsx:81]: if i just _state: etc etc etc

Comment (mattwilson02) [reply src/Screens/MealOverview.tsx:81]: Why do u need to pass any parameters if you aren't using them?

File: src/Screens/MealOverview.tsx [hunk 4]
```diff
@@ -60,7 +62,12 @@ const styles = StyleSheet.create({
     width: "100%"
   }
 });
-//css here is kinda janky fixing it now
+//css still janky will fix in the css pr
+
+type SequenceValue = {
+  state: Meal[];
+};
```
Comment (mattwilson02) [inline src/Screens/MealOverview.tsx:69]: ok now you can remove that type too

File: src/Screens/MealOverview.tsx [hunk 5]
```diff
@@ -70,6 +77,22 @@ export const MealOverview = () => {
     mealsOutDiet: state.meals.filter(item => !item.inDiet)
   }));
 
+  const countConsecutiveTrue = () => {
```
Comment (mattwilson02) [inline src/Screens/MealOverview.tsx:80]: technically you dont even need a function here, now that you have two variables, consecutive count and maxcount, you can just use that to display your value

File: src/Screens/MealOverview.tsx [hunk 6]
```diff
@@ -70,6 +73,20 @@ export const MealOverview = () => {
...
+
+  for (const item of allMeals) {
+    if (item === true) {
+      consecutiveCount++;
+      maxConsecutiveCount = Math.max(maxConsecutiveCount, consecutiveCount);
+    } else {
+      consecutiveCount = 0;
+    }
+  }
+  // return maxConsecutiveCount;
```
Comment (mattwilson02) [inline src/Screens/MealOverview.tsx:88]: remove that comment

Comment (JakeBalakrishna) [reply src/Screens/MealOverview.tsx]: done

**Review by mattwilson02 — APPROVED** [review]

**Pull request merged**